src/ray/raylet/worker_pool.cc

```diff
@@ -138,7 +138,9 @@ void WorkerPool::StartWorkerProcess(const Language &language) {
   int rv = execvp(worker_command_args[0],
                   const_cast<char *const *>(worker_command_args.data()));
   // The worker failed to start. This is a fatal error.
-  RAY_LOG(FATAL) << "Failed to start worker with return value " << rv;
+  if (rv < 0) {
+    RAY_LOG(FATAL) << "Failed to start worker: " << strerror(errno);
```

Comment:
Yes, I did meet this case in practice.

**Detail:**
if there was no JDK in my `$PATH`, `StartWorker()` couldn't find the `java` executable file.
The log printed into my log file is:
```shell
F0929 10:05:07.341799 1929011200 worker_pool.cc:142] Failed to start worker: -1
  2 *** Check failure stack trace: ***
  3     @        0x10481ebca  google::LogMessage::Fail()
  4     @        0x10481caaa  google::LogMessage::SendToLog()
  5     @        0x10481d8af  google::LogMessage::Flush()
  6     @        0x10481d6e9  google::LogMessage::~LogMessage()
  7     @        0x10481d9a5  google::LogMessage::~LogMessage()
  8     @        0x10478a6fd  ray::RayLog::~RayLog()
  9     @        0x1047b151b  ray::raylet::WorkerPool::StartWorkerProcess()
 10     @        0x1047cfaf3  ray::raylet::NodeManager::AssignTask()
 11     @        0x1047cf21b  ray::raylet::NodeManager::DispatchTasks()
 12 F0929 10:05:07.342034 1929011200 worker_pool.cc:142] Failed to start worker: -1
 13     @        0x1047d5f50  ray::raylet::NodeManager::EnqueuePlaceableTask()
```
Because the return value  of `execvp` simply means whether it failed, regardless of the failure reason,`rv` is always `-1`.

Em. we can't get the reason for failure according this log message.

After this change, the log message will be:
```shell
F0929 10:24:20.733393 1929011200 worker_pool.cc:142] Failed to start worker: No such process
  2 *** Check failure stack trace: ***
  3 F0929 10:24:20.733690 1929011200 worker_pool.cc:142] Failed to start worker: No such process
  4 *** Check failure stack trace: ***
  5     @        0x107eaabca  google::LogMessage::Fail()
  6     @        0x107eaabca  google::LogMessage::Fail()
  7     @        0x107ea8aaa  google::LogMessage::SendToLog()
```
It help us to find the failure reason more simply.

Reply:
In my practice case, `fork()` is ok.

Reply:
in my view, for the robustness, we should add ERROR log when `fork()` fails.
If it's necessary, I can add it in this PR.